openedx_events/bridge/avro_attrs_bridge.py

```diff
@@ -139,11 +139,12 @@ def _generate_avro_record_for_attrs_class(
         """
         field: Dict[str, Any] = {}
         field["name"] = field_name
-        field["type"] = dict(name=attrs_class.__name__, type="record", fields=[])
+        record_type = dict(name=attrs_class.__name__, type="record", fields=[])
+        field["type"] = record_type
```

Comment:
Nit: My personal preference would be to move this line right before you return field, and after you have manipulated record_type. Just a preference though.